Allow configuring rule syntax with Kubernetes Ingress annotation (#10985)
traefik/traefik: docs/content/migration/v2-to-v3-details.md

All Pilot related configuration should be removed from the static configuration.

### Kubernetes Ingress Path Matching

In v3, the Kubernetes Ingress default path matching does not support regexes anymore.

#### Remediation

> Two level of remediation are possible:
> 
> - Interpret the default path matcher `PathPrefix` with v2 syntax.
> It can done globally (for all routers) with the [static configuration](#configure-the-default-syntax-in-static-configuration) or per router by using the specific annotation: [traefik.ingress.kubernetes.io/router.rulesyntax](../routing/providers/kubernetes-ingress.md#annotations)
> 
> - Adapt the path regular expression to be compatible the go regex syntax and change the default path matcher for the `PathRegexp` matcher with the annotation `traefik.ingress.kubernetes.io/router.pathmatcher`.

Reviewer: ```suggestion
Two levels of remediation are possible:

- Interpret the default path matcher `PathPrefix` with v2 syntax.
This can done globally for all routers with the [static configuration](#configure-the-default-syntax-in-static-configuration) or on a per-router basis by using the [traefik.ingress.kubernetes.io/router.rulesyntax](../routing/providers/kubernetes-ingress.md#annotations) annotation.

- Adapt the path regex to be compatible with the Go regex syntax and change the default path matcher to use the `PathRegexp` matcher with the [`traefik.ingress.kubernetes.io/router.pathmatcher`](../routing/providers/kubernetes-ingress.md#annotations) annotation.
```

> - Adapt the path regular expression to be compatible the go regex syntax and change the default path matcher for the `PathRegexp` matcher with the annotation `traefik.ingress.kubernetes.io/router.pathmatcher`.

Reviewer: Maybe we can change the pathmatcher annotation documentation to indicate that the `PathRegex` matcher can be used, here: https://doc.traefik.io/traefik/routing/providers/kubernetes-ingress/.

Author: I have modified the annotation documentation in that page already, I have added the mention of the `PathRegexp` matcher, don't you think it is enough?
